sdk/azidentity/azidentity.go

```diff
@@ -30,23 +32,17 @@ const (
 
 const azureAuthorityHost = "AZURE_AUTHORITY_HOST"
 
-// AuthorityHost is the base URL for Azure Active Directory
-type AuthorityHost string
-
-const (
-	// AzureChina is a global constant to use in order to access the Azure China cloud.
-	AzureChina AuthorityHost = "https://login.chinacloudapi.cn/"
-	// AzureGovernment is a global constant to use in order to access the Azure Government cloud.
-	AzureGovernment AuthorityHost = "https://login.microsoftonline.us/"
-	// AzurePublicCloud is a global constant to use in order to access the Azure public cloud.
-	AzurePublicCloud AuthorityHost = "https://login.microsoftonline.com/"
-)
-
-// setAuthorityHost initializes the authority host for credentials.
-func setAuthorityHost(authorityHost AuthorityHost) (string, error) {
-	host := string(authorityHost)
+// setAuthorityHost initializes the authority host for credentials. Precedence is:
+// 1. cloud.Configuration.LoginEndpoint value set by user
+// 2. value of AZURE_AUTHORITY_HOST
+// 3. default: Azure Public Cloud
+func setAuthorityHost(cc cloud.Configuration) (string, error) {
+	host := cc.LoginEndpoint
 	if host == "" {
-		host = string(AzurePublicCloud)
+		if !reflect.ValueOf(cc).IsZero() {
```

Comment:
The only other field to check here is if `Services` is nil, right? Do we need reflection for that?

Reply:
Good point! Checking whether `Services` is empty would suffice.